Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -1,7 +1,144 @@
+            return operand2;
+        }
+        public string getExpectedOperand2(){
+            return expectedOperand2;
+        }
+        private void setOperation(string value){
+            operation=value;
+        }
+        public string getOperation(){
+            return operation;
+        }
+        private void setResult(string value){
+            result=value;
+        }
+        public string getResult(){
+            return result;
+        }
+        public string getExpectedResult(){
+            return expectedResult;
+        }
+        private void setExpectedValues(){
+            if(operation=="*"){
+                //expectedOperand1
+                if(!operand1.Contains('?'))
+                expectedOperand1=operand1;
+                else{
+                    var operand2Int=Convert.ToInt32(operand2);
+                    var resultInt=Convert.ToInt32(result);
+                    if(operand2Int==0)
+                    expectedOperand1="infinite";
+                    else if(resultInt%operand2Int!=0)
+                    expectedOperand1="unexpected";
+                    else{
+                        expectedOperand1=(resultInt/operand2Int).ToString();
+                    }
+                }
+                //expectedoperand2
+                if(!operand2.Contains('?'))
+                expectedOperand2=operand2;
+                else{
+                    var operand1Int=Convert.ToInt32(operand1);
+                    var resultInt=Convert.ToInt32(result);
+                    if(operand1Int==0)
+                    expectedOperand2="infinite";
+                    else if(resultInt%operand1Int!=0)
+                    expectedOperand2="unexpected";
+                    else{
+                        expectedOperand2=(resultInt/operand1Int).ToString();
+                    }
+                }
+                //expected result
+                if(!result.Contains('?'))
+                expectedResult=result;
+                else{
+                    var operand1Int=Convert.ToInt32(operand1);
+                    var operand2Int=Convert.ToInt32(operand2);
+                    expectedResult=(operand1Int*operand2Int).ToString();
+                }
+            }
+        }
+        public Equation(string operand1,string operand2,string operation,string result){
+            setOperand1(operand1);
+            setOperand2(operand2);
+            setOperation(operation);
+            setResult(result);
+            setExpectedValues();
+        }
+    }
+    class FixMultiplication{
+        public static int FindDigit(string equation){
+            Equation thisEquation=ParseEquation(equation);
+            return CorrectEquationAndReturnMissingDigit(thisEquation);
+        }
+        private static int CorrectEquationAndReturnMissingDigit(Equation equation){
+            if(equation.getOperand1().Contains('?'))
+            return CompareAndFind(equation.getOperand1(),equation.getExpectedOperand1());

```

Comment:
Rename this method to CorrectAndReturnMissingDigit and calculate expected value there only